### PR TITLE
Search project names endpoint impl

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java [hunk 1]
```diff
@@ -337,6 +337,16 @@ public ProjectInfoResource getProjectInfo(@PathVariable String projectName,
 		return getProjectHandler.getAllProjectNames();
 	}
 
+	@Transactional(readOnly = true)
+	@PreAuthorize(ADMIN_ONLY)
+	@RequestMapping(value = "/names/search", method = RequestMethod.GET)
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java]: @GetMapping

File: src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java [hunk 2]
```diff
@@ -337,6 +337,16 @@ public ProjectInfoResource getProjectInfo(@PathVariable String projectName,
...
+	@ResponseBody
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java]: it is @RestController so we don't need this annotation

File: src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java]: fixed

File: src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/ws/controller/ProjectController.java]: fixed

**Pull request merged**